src/docsfy/postprocess.py

```
@@ -15,12 +16,135 @@
+
+    updated: dict[str, str] = {}
+    for slug, content in pages.items():
+
+        def _replace_link(match: re.Match[str], _slug: str = slug) -> str:
+            link_text = match.group(1)
+            target_slug = match.group(2)
+            if target_slug.lower() in valid_slugs_lower:
+                return match.group(0)  # Keep valid links
+            logger.info(
+                f"[{_label}] Removing broken link to '{target_slug}.html' "
+                f"in page '{_slug}'"
```

Ruff (0.15.10)

[warning] Logging statement uses f-string

(G004)